### ISSUE TITLE
Cannot use multiple Segment integrations on the same website 

### ISSUE THREAD

**Issue by Vadorequest** [segmentio/analytics-next#785]
We use no-code tools, which already use Segment for their own purposes.

Because of that, we are stuck and can't use Segment for our own purposes, because the script attaches things to `window.analytics` and it is not possible to have multiple Segment integrations within the same page.

We really need a way to work around this.

Also, other libraries use `window.analytics` and they're therefore completely incompatible with Segment.
I believe it is possible to kill two birds with one rock here, and solve both issues at once.

It seems Segment had a "no conflict" option in v1, but it has been deprecated in v2, and it's only usable when self-hosting. (which is not possible in our case, and the no-code service we use don't self-host either)

**Comment by ohemi-newrelic** [comment]
+1
Is this being handled?
There should be an option to not attach the analytics instance to the window

**Comment by NicoleG25** [comment]
+2
This is really affecting us. Could you provide a solution/a workaround?

@nettofarah @silesky (sorry for the random tag but we're in a tough spot)
please let me know if there is anyone else I should talk to, to get this moving along :)

**Comment by silesky** [comment]
@NicoleG25 
Sorry, this isn't on the roadmap for this quarter and I can't give any timeline -- we have a lot on our plate, and are short staffed ATM. The main issue is localStorage conflict and the architecture -- if someone wanted to dig into the library and attempt to implement this, that would certainly be OK, but there are some complex edge cases. 

If you don't have device mode plugins, you can use `@segment/analytics-node` with `maxEventsInBatch: 1` which should work in any runtime where fetch is available. https://github.com/segmentio/analytics-next/tree/master/packages/node

Here's another similar issue: https://github.com/segmentio/analytics-next/issues/367

**Comment by silesky** [comment]
@Eunanibus

We now support renaming global analytics, so you can have different write keys for each instance.

https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#installing-the-library-under-a-custom-global-namespace

However, if I'm understanding you correctly -- for microfrontends, unlike low-code, you have the option of just using the npm version of the library.

**Comment by silesky** [comment]
> Also, other libraries use window.analytics and they're therefore completely incompatible with Segment.
I believe it is possible to kill two birds with one rock here, and solve both issues at once.

We actually do support renaming the global analytics instances now, so you can have analytics1 and analytics2. Should *mostly* work -- https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#installing-the-library-under-a-custom-global-namespace

**Comment by Eunanibus** [comment]
Yep, I found this too, it was very buried but it **is** documented in the code.

This appeared to work fine, and instead of altering the parent container for a different namespace, I found that I could just adjust the child container namespace instead.

This (appears?) to have completely eliminated the conflict. So for anyone who maybe finds this in the future and is struggling with multiple Segment instances in a Micrfrontend (or Parent Child) architecture, make use of the `globalAnalyticsKey` option field to isolate them.
